### PR TITLE
Material support; fixes #3

### LINKED ISSUE #3: DST-CAT-SR-0011: Material Properties Mapping
The mapping shall support transfer of  material properties without loss of information.

-    Find the data CATIA object structure
-    Map a SampledFunction (Face/Surface ? id, Material) ParameterType in a UI. Every Face/Surface Id should be in the list.

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEH-CATIA/pulls) open
- [x] I have verified that I am following the DEH-CATIA [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEH-CATIA/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

; fixes #3 
- [x] Materials information on Product and Body can be transfered to the Hub as well as to Catia.
- [x]  When mapping from the Hub, it is possible to remove materials as well
- [x]  Mapping gets reloaded on session end update
- [x] Added the bodies row in all the product trees as well as a Material column
- [x] fixed some bugs in loading the saved mapping
- [x] Minor ui improvements

### CODE REVIEW

**Comment by nathanatrhea** [comment]
image

File: DEHCATIA.Tests/MappingRules/ElementToCatiaRuleTestFixture.cs [hunk 1]
```diff
@@ -167,9 +169,23 @@ public void Setup()
...
                         }
+                    },
+                    new Parameter()
+                    {
+                        ParameterType = new SampledFunctionParameterType(),
+                        ValueSet =
+                        {
+                            new ParameterValueSet()
+                            {
+                                Manual = new ValueArray<string>(new List<string>() { "body.0", "material0" }),
```
Comment (alexatrhea) [inline DEHCATIA.Tests/MappingRules/ElementToCatiaRuleTestFixture.cs:180]: maybe a more extensive test with more materials and bodies required?

File: DEHCATIA.Tests/Services/ComConnector/CatiaComServiceTestFixture.cs [hunk 2]
```diff
@@ -145,7 +149,7 @@ public void VerifyUpdateElement()
             this.element.CatiaElement.Parameters.Add(new StringParameterViewModel(massParameter.Object, "56.4"));
 
             this.element.CatiaElement.Shape = this.element.CatiaElement.Parameters.GetShape();
-            Assert.IsTrue(this.element.CatiaElement.Shape.IsSupported);
+            Assert.IsFalse(this.element.CatiaElement.Shape.IsSupported);
```
Comment (alexatrhea) [inline DEHCATIA.Tests/Services/ComConnector/CatiaComServiceTestFixture.cs:152]: why false all of a sudden?

File: DEHCATIA.Tests/Services/ComConnector/CatiaComServiceTestFixture.cs [hunk 3]
```diff
@@ -176,7 +180,7 @@ public void VerifyCreateElement()
 
             this.element.CatiaElement.Shape = this.element.CatiaElement.Parameters.GetShape();
             File.Copy(Path.Combine(TestContext.CurrentContext.TestDirectory, TemplatePartName), destFileName);
-            Assert.IsTrue(this.element.CatiaElement.Shape.IsSupported);
+            Assert.IsFalse(this.element.CatiaElement.Shape.IsSupported);
```
Comment (alexatrhea) [inline DEHCATIA.Tests/Services/ComConnector/CatiaComServiceTestFixture.cs:183]: same?

File: DEHCATIA.Tests/Services/MappingConfiguration/MappingConfigurationServiceTestFixture.cs [hunk 4]
```diff
@@ -229,12 +233,13 @@ public void VerifyLoadMappingFromHubToDst()
...
-            
-           this.hubController.Setup(x => x.GetThingById(It.IsAny<Guid>(), It.IsAny<Iteration>(), out element)).Returns(true);
+
+            element = this.element0;
+            this.hubController.Setup(x => x.GetThingById(It.IsAny<Guid>(), It.IsAny<Iteration>(), out element)).Returns(true);
            Assert.DoesNotThrow(() => mappedRows = this.service.LoadMappingFromHubToDst(this.elements));
 
-           this.hubController.Verify(x => x.GetThingById(It.IsAny<Guid>(), It.IsAny<Iteration>(), out element), Times.Exactly(3)); 
-           Assert.AreEqual(1, mappedRows.Count);
+           this.hubController.Verify(x => x.GetThingById(It.IsAny<Guid>(), It.IsAny<Iteration>(), out element), Times.Exactly(0));
```
Comment (alexatrhea) [inline DEHCATIA.Tests/Services/MappingConfiguration/MappingConfigurationServiceTestFixture.cs:241]: why the changes in the tests?

File: DEHCATIA/Enumerations/ShapeKind.cs [hunk 5]
```diff
@@ -58,27 +58,7 @@ public enum ShapeKind
         /// Shape kind is a Paraboloid
         /// </summary>
         Paraboloid,
-
-        /// <summary>
-        /// Shape kind is a Triangular Prism
-        /// </summary>
-        TriPrism,
```
Comment (alexatrhea) [inline DEHCATIA/Enumerations/ShapeKind.cs:65]: Whey these shapes removed?

**Review by alexatrhea — COMMENTED** [review]
Some questions

**Comment by nathanatrhea** [comment]
image
